File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseBetween(struct ListNode* head, int left, int right) {
    struct ListNode*dummy=(struct ListNode*)malloc(sizeof(struct ListNode));
    dummy->next=head;
    struct ListNode*leftpre=dummy;
    struct ListNode*current=head;
    struct ListNode*prev=NULL;
    for(int i=1;i<left;i++){
        current=current->next; 
         leftpre=leftpre->next;
    }
    struct ListNode*sublisthead=current;
    int n=right-left+1;
    for(int i=0;i<n;i++){
        struct ListNode*nextn=current->next;
        current->next=prev;
        prev=current;
        current=nextn;
    }
    sublisthead->next=current;
    leftpre->next=prev;
    
    return dummy->next;   
}

File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseBetween(ListNode head, int left, int right) {
       ListNode dummy = new ListNode(0);
        dummy.next = head;

        ListNode beforeReverse = dummy; 
        ListNode current = head;        

        for (int i = 0; i < left - 1; i++) {
            current = current.next;
            beforeReverse = beforeReverse.next;
        }

        ListNode startOfReverse = current; 
        ListNode previous = null;          

        for (int i = 0; i < (right - left) + 1; i++) {
            ListNode nextNode = startOfReverse.next;
            startOfReverse.next = previous;
            previous = startOfReverse;
            startOfReverse = nextNode;
        }
        beforeReverse.next = previous;
        current.next = startOfReverse;

        return dummy.next;
        
    }
}